### PR TITLE
Add ShortCircuitBlockWriteHandlerTest

### PR DESCRIPTION
**Pull request by YangchenYe323**
### What changes are proposed in this pull request?
Added unit tests for `ShortCircuitBlockWriteHandler`

### Why are the changes needed?
Improved unit test coverage(0% -> 81%)

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+    public List<CreateLocalBlockResponse> getResponses() {
+      return ImmutableList.copyOf(mReceivedResponses);
+    }
+
+    public Throwable getError() {
+      return mError;
+    }
+
+    public boolean ismCompleted() {
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:194]:
```suggestion
    public boolean isCompleted() {
```

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    assertFalse(mBlockWorker.isBlockCommitted(blockId));
+
+    mHandler.onCompleted();
+
+    // verify that the block is committed and publicly available
+    assertTrue(mBlockWorker.isBlockCommitted(blockId));
+  }
+
+  @Test
+  public void nextRequestWithoutCleaningUp() {
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:92]:
```suggestion
  public void nextRequestWithoutCommitting() {
```

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+    // the session cleaned up
+    assertFalse(mBlockWorker.isTempBlockCreated(blockId));
+    assertFalse(mBlockWorker.isBlockCommitted(blockId));
+
+    // verify that we get the correct response
+    assertFalse(mResponseObserver.ismCompleted());
+    assertNotNull(mResponseObserver.getError());
+  }
+
+  private CreateLocalBlockRequest createRequest(long blockId, boolean onlyReserveSpace) {
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:156]: Inline this method.

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    mHandler.onNext(anotherRequest);
+
+    assertNotNull(mResponseObserver.getError());
+    assertFalse(mResponseObserver.ismCompleted());
+    assertFalse(mBlockWorker.isTempBlockCreated(blockId));
+    assertFalse(mBlockWorker.isTempBlockCreated(anotherBlockId));
+  }
+
+  @Test
+  public void requestSpaceForNonExistingBlock() {
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:108]:
```suggestion
  public void cannotReserveSpaceForNonExistingBlock() {
```

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+    // is not created yet
+    mHandler.onNext(request);
+
+    assertTrue(mResponseObserver.getResponses().isEmpty());
+    assertFalse(mResponseObserver.ismCompleted());
+    Throwable t = mResponseObserver.getError();
+    assertTrue(
+        t != null
+        && t.getMessage().contains("does not exists"));
+  }
```
Comment (dbw9580) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:121]: I think we do not have to inspect the exception. Only assert that an exception is thrown.

**Review by dbw9580 — CHANGES_REQUESTED** [review]

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java (hunk 1 above)
Comment (YangchenYe323) [reply core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java]: Done

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java (hunk 3 above)
Comment (YangchenYe323) [reply core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java]: Fixed

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java (hunk 4 above)
Comment (YangchenYe323) [reply core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java]: Fixed

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java (hunk 5 above)
Comment (YangchenYe323) [reply core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java]: Fixed

**Comment by YangchenYe323** [comment]
@dbw9580 PTAL!

**Review by dbw9580 — APPROVED** [review]
LGTM. @beinan would you like to take another look?

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+    mBlockWorker = new TestBlockWorker(tmpDir.getAbsolutePath());
+
+    mResponseObserver = new TestResponseObserver();
+
+    mHandler = new ShortCircuitBlockWriteHandler(mBlockWorker, mResponseObserver);
+  }
+
+  @Test
+  public void createBlock() {
+    long blockId = 1L;
```
Comment (beinan) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:65]: nit: make a constant for this blockId?

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+    CreateLocalBlockRequest request = CreateLocalBlockRequest
+            .newBuilder()
+            .setBlockId(blockId)
+            .build();
```
Comment (beinan) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:69]: looks like you're creating this request again and again,  how about construct it in the before() method?

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+  @Test
+  public void cannotReserveSpaceForNonExistingBlock() {
+    long blockId = 1L;
+    CreateLocalBlockRequest request = CreateLocalBlockRequest
+            .newBuilder()
+            .setBlockId(blockId)
+            .setOnlyReserveSpace(true)
+            .build();
+    // this request should fail as the block
+    // is not created yet
```
Comment (beinan) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:125]: is the line too long if we concat these two line together?

File: core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+
+    @Override
+    public String createBlock(long sessionId, long blockId, int tier,
+                              CreateBlockOptions createBlockOptions) throws IOException {
+      if (mTempBlocks.containsKey(blockId) && mPublicBlocks.contains(blockId)) {
+        throw new RuntimeException(String.format("Block %d already exists", blockId));
+      }
+      mTempBlocks.put(blockId, sessionId);
+      String filePath = getPath(blockId);
+      // create file
```
Comment (beinan) [inline core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java:259]: I think this comment is unnecessary, what do you think?

**Review by beinan — COMMENTED** [review]
Good job!  it looks good to me except a few minor style issue.

**Comment by YangchenYe323** [comment]
@beinan Made an update to address your comment. PTAL!

**Comment by YangchenYe323** [comment]
@beinan Thanks for the comment! Made an update that addressed your comment. PTAL.

**Review by beinan — APPROVED** [review]
lgtm

**Pull request merged**